kubert/src/runtime.rs

```diff
@@ -295,11 +386,65 @@ impl<S> Runtime<S> {
         T::DynamicType: Default,
     {
         let watch = watcher::watcher(api, watcher_config);
+        #[cfg(feature = "prometheus-client")]
+        let watch = self.instrument_watch(watch);
         let successful = errors::LogAndSleep::fixed_delay(self.error_delay, watch);
         let initialized = self.initialized.add_handle().release_on_ready(successful);
         shutdown::CancelOnShutdown::new(self.shutdown_rx.clone(), initialized)
     }
 
+    #[cfg(feature = "prometheus-client")]
+    fn instrument_watch<T>(
+        &self,
+        watch: impl Stream<Item = watcher::Result<watcher::Event<T>>> + Send,
+    ) -> impl Stream<Item = watcher::Result<watcher::Event<T>>> + Send
```

Comment:
Notice how, for the other combinators (shutdown, log and sleep, etc) we keep those implementations outside of the runtime module, which includes high-level instrumentation combinators.

We should be able to take the same approach here. `instrument_watch` needs only the ResourceWatchMetrics and no other part of the runtime. I would probably create a sub-module in `kubert/src/runtime/watch_metrics.rs` that includes a 
 `pub(super) struct ResourceWatchMetrics` and a `pub(super) fn instrument(metrics: ResourceWatchMetrics, watch: impl ...)`
 
The the runtime module can feature-gate the entire module and reduce boilerplate. This will also make it easier to read the metrics code distinctly.